python/cudaq/operators/helpers.py

```diff
@@ -0,0 +1,115 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import inspect, numpy, os, re, sys, typing  # type: ignore
+from typing import Any, Callable, Iterable, Mapping, Optional, Sequence, Tuple
+
+if (3, 11) <= sys.version_info:
+    NumericType = typing.SupportsComplex
+elif (3, 10) <= sys.version_info:
+    # type | type syntax is only supported in 3.10 or later
+    NumericType = numpy.complexfloating | complex | float | int
+else:
+    NumericType = typing.Union[numpy.complexfloating, complex, float, int]
```

Comment:
Perhaps we can now remove this `else` block since we don't support Python 3.9 anymore?